File: libraries/valid-email2/validate.rb
#!/usr/bin/env ruby
require 'valid_email2'

if ARGV.length < 1
  STDERR.puts "Usage: validate.rb <addresslist.txt>"
  exit 1
end

File.readlines(ARGV[0]).each do |line|
  email = line.chomp
  next if email.empty?

  result = ValidEmail2::Address.new(email).valid? ? "valid   " : "invalid "
  puts "#{result}#{email}"
end

File: libraries/email-valid-perl/validate.pl
#!/usr/bin/env perl
use strict;
use warnings;
use Email::Valid;

my $file = $ARGV[0] or die "Usage: validate.pl <addresslist.txt>\n";
open my $fh, '<', $file or die "Cannot open $file: $!\n";

while (my $line = <$fh>) {
    chomp $line;
    next if $line eq '';

    my $valid = Email::Valid->address($line);
    my $result = $valid ? "valid   " : "invalid ";
    print "$result$line\n";
}

close $fh;
